utils/message_utils.js

```diff
@@ -26,57 +26,75 @@ let cmakePaths = cmakePath.split(':');
 let jsMsgPath = 'share/node_js/ros';
 
 let messagePackageMap = {};
+let messagePackagePathMap = {};
 
 //-----------------------------------------------------------------------
 // Utilities for loading, finding handlers for
 // message serialization/deserialization
 //-----------------------------------------------------------------------
 
 let MessageUtils = {
```

Comment:
Can you give a brief explanation (in comments) of how the message finding / caching / loading works for rosjs?